src/QMCHamiltonians/tests/test_ion_derivs.cpp

```diff
@@ -769,9 +770,9 @@ TEST_CASE("Eloc_Derivatives:proto_sd_noj", "[hamiltonian]")
   REQUIRE(psi != nullptr);
   //end incantation
 
-//  TWFPrototype twf;
+  TWFPrototype twf;
 
-//  psi->initialize_TWF_Prototype(elec, twf);
+  psi->initialize_TWF_Prototype(elec, twf);
```

Comment:
I do like the idea of a factory, but I chose to do it this way for two reasons.  1.) This object is designed to be completely dependent on a properly constructed TrialWaveFunction object.  TrialWaveFunction always exists, whereas this is a wrapper that is constructed if force derivatives need to be computed.  2.) If I don't want to rebuild the whole wave function from scratch, I need some private TrialWaveFunction tags to determine if things are Jastrows or Determinants.  Having psi know how to construct it's own fancy wrapper seemed to be the simpler solution for me at the time.

Reply:
As @rcclay has mentioned, this is a wrapper intended to be used only for the force derivatives codes he is pushing upstream. So this will be contained instead of being part of the WaveFunctionFactory if it is the factory you mean. We will explore routes without the need of wrapper but this what we have now.

Reply:
`getOrbitals()` should not be widely used except for testing purpose. Don't assume the internal structure of TWF. For this reason, my choice will be the current code.

Reply:
Having the more basic type TrialWaveFunction have knowledge of extending type  is not scalable. TrialWaveFunction is a compute interface not an initializer for other objects. This register call is properly part of how a TWFPrototype becomes valid, it should be with the implementation of TWFPrototype.

It should preferably be called from the constructor of TWFPrototype.
I can see there is a similar issue pending for the register_TWF_prototype call on the WFC but I'm waiting to see what that looks like.  

The important news here is that the actual data dependence that TWFPrototype has is on the fermionic wavefunction components it shares with TWF it makes sense for that to be expressed in the code for TWFPrototype.

Reply:
Since WFC are virtual, its iteration needs to be implemented on the WFC side and hence `register_TWF_Prototype()` calls.

> It should preferably be called from the constructor of TWFPrototype.

Agree
```
TWFPrototype twf_proto(twf);
```
is nicer than directly calling initialize_TWF_Prototype.